src/libsyntax/feature_gate.rs

```diff
@@ -385,6 +386,13 @@ impl<'a> Context<'a> {
                 return;
             }
         }
+        for &(ref n, ref ty) in self.plugin_attributes.iter() {
+            if &*n == name {
+                // Plugins can't gate attributes, so we don't check for it
```

Comment:
Isn't it more that plugins register "known" attributes, i.e. they no longer count as custom?